internal/proto/workflow/v2/workflow.proto

```diff
@@ -0,0 +1,117 @@
+syntax = "proto3";
+
+package internal.proto.workflow.v2;
+
+option go_package = "github.com/tinkerbell/tink/internal/proto/workflow/v2;workflow";
+
+// WorkflowService is responsible for retrieving workflows to be executed by the agent and
+// publishing events as a workflow executes.
+service WorkflowService {
+  // Stream creates a stream that will receive workflows intended for the agent identified
+  // by the StreamWorkflowsRequest.agent_id.
+  rpc StreamWorkflows(StreamWorkflowsRequest) returns (stream StreamWorkflowsResponse) {}
```

Comment:
thought this was interesting. Its from the Buf website. https://buf.build/docs/best-practices/style-guide#recommendations 

_"While controversial, our recommendation is to avoid streaming RPCs. While they certainly have specific use cases that make them extremely valuable, on the whole they generally cause a lot of problems, push RPC framework logic up the stack, and usually prevent developing more reliable architectures."_

Reply:
Also, would you me open to a name change here? maybe `StreamWorkflows` -> `GetWorkflows`

Reply:
> thought this was interesting. Its from the Buf website. [buf.build/docs/best-practices/style-guide#recommendations](https://buf.build/docs/best-practices/style-guide#recommendations)

I'm familiar with this but chose to ignore it because the alternatives I thought about have tradeoffs also. The options I came up with include: (1) polling from the agent to the server for workflows; (2) launching a server in the agent and have the tink-server (now a confusing name) issue unary calls to run workflows. 

I wasn't fond of polling even with HTTP2s sticky connections and I wanted to, initially at least, avoid the complexity of launching a server and dealing with the agent going offline etc. 

I very open to adopting a different approach and welcome more ideas. Long term I'm interested in migrating to the agent containing a server even though its more complex.

> Also, would you me open to a name change here? maybe StreamWorkflows -> GetWorkflows

Yes, no problem.